src/status/libstatus/wallet.nim

```diff
@@ -70,6 +73,38 @@ proc sendTransaction*(tx: EthSend, password: string): RpcResponse =
 
   trace "Transaction sent succesfully", hash=result
 
+proc sendTransaction*(tokens: JsonNode, source, to, assetAddress, value, gas, gasPrice, password: string): RpcResponse =
+  var
+    weiValue = eth2Wei(parseFloat(value), 18) # ETH
+    data = ""
+    toAddr = eth_utils.parseAddress(to)
+  let gasPriceInWei = gwei2Wei(parseFloat(gasPrice))
+
+  # TODO: this code needs to be tested with testnet assets (to be implemented in
+  # a future PR
+  if assetAddress != constants.ZERO_ADDRESS and not assetAddress.isEmptyOrWhitespace:
+    let
+      token = tokens.first("address", assetAddress)
+      contract = contractMethods.getContract("snt")
+      transfer = coder.Transfer(to: toAddr, value: eth2Wei(parseFloat(value), token["decimals"].getInt))
+    weiValue = 0.u256
+    let transferThing = contract.methods["transfer"]
+    data = transferThing.encodeAbi(transfer)
+    toAddr = eth_utils.parseAddress(assetAddress)
+
+  let tx = EthSend(
+    source: eth_utils.parseAddress(source),
+    to: toAddr.some,
+    gas: (if gas.isEmptyOrWhitespace: Quantity.none else: Quantity(cast[uint64](parseFloat(gas).toUInt64)).some),
+    gasPrice: (if gasPrice.isEmptyOrWhitespace: int.none else: gwei2Wei(parseFloat(gasPrice)).truncate(int).some),
+    value: weiValue.some,
+    data: data
+  )
+  try:
+    result = sendTransaction(tx, password)
+  except RpcException as e:
+    raise
+
```

Comment:
Ah, just saw that you modified it as well. The reason why I moved the function in libstatus is that to use `spawnAndSend`, you **cannot** use `self`, so it was impossible to call the wallet model. So I made `sendTransaction` static so that it doesn't rely on being an instance anymore. 

We can just move the changes you added to `sendTransaction` to this new function here. If you have another idea or if maybe I misunderstood `spawenAndSend`, please tell me because I do agree that it's not ideal.

Reply:
Turns out I was dumb and it is possible to use the wallet model, so I reverted my change and used your functions inside the spawnAndSend